File: algorithms/[406]根据身高重建队伍/solution.js
/**
 * @param {number[][]} people
 * @return {number[][]}
 */
const reconstructQueue = function(people) {
  if (!people || !people.length) return [];

  const res = [];

  people.sort((a, b) => {
    if (a[0] === b[0]) return a[1] - b[1];
    return b[0] - a[0];
  }).forEach(item => res.splice(item[1], 0, item));

  return res;
};

File: algorithms/[406]根据身高重建队伍/solution.py
# topics = ["贪心算法"]

from typing import List


class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        res: List[List[int]] = []

        # 原数组按第一个元素降序，第二个元素升序
        people.sort(key=lambda ele: (-ele[0], ele[1]))

        # 按第二个元素插入排序
        for h, k in people:
            res.insert(k, [h, k])

        return res
